### PR TITLE
New tooling and test refactoring

### PR DESCRIPTION
**Pull request by Dresdn**
(Provide a general summary of your changes in the Title above)

## Description
* What is the current behavior? You can also link to an open issue here.

This is for #69 and implements poetry for management, as well as refactors tests to use pytest.

All changes are described in the various commits.

#### Styling Improvements

I did add a few things to make the code style more clean and more "best practices" by implementing some packages:
* `flake8`
* [black](https://github.com/psf/black)
* [wemake-python-styleguide](https://wemake-python-stylegui.de)

The `wemake-python-styleguide` is pretty strict, but it's one of the best I've come across to drive good development practices to make maintainability easier. I get if it's too strict, but I don't mind going through and cleaning it up a bit.

#### Tests

I did a very first pass at converted to pytest, strictly by just using a new test django app that's separate from the `/tests` directory. There's a lot of work that needs to be done as there are some errors introduced when randomizing the tests (red flag), and test coverage is only ~85%.

## Motivation and Context
* Why is this change required? What problem does it solve?

Various. The project uses tooling that hasn't been updated in quite some time, and management of a development environment is fairly difficult. This approach should make overall maintenance of the project much easier and more consistent.

## How Has This Been Tested?
* Did you add appriopriate tests?

All tests pass ... mostly. When introducing pytest-randomly, I noticed a test will fail once in a while, so more work needs to be done to refactor the tests themselves.

* Please describe in detail how you tested your changes.

Run:
`poetry install`
`poetry pytest` or `poetry shell`; `pytest`

## Next Steps

Should this be an agreed upon direction, next steps would be to implement `poetry publish` Github actions to automagically publish to pypi, run test coverage on PRs, and other CI best practices.

Appreciate all feedback!

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,32 +1,17 @@
 [tox]
+isolated_build = true
 envlist =
-    py36-django{31,32,tip},
-    py37-django{31,32,tip},
-    py38-django{31,32,tip}
-    py39-django{31,32,tip}
-    migrationscheck
+    py{35,36,37,38,39}-django22
```
Comment (Dresdn) [inline tox.ini]: I forgot to mention, I did add back in testing for Django 2.2 as it's LTS and should be something the project supports through 1H 2022.

**Comment by lvm** [comment]
This is wonderful, i'll take a look at this this weekend

Thanks!

**Comment by Dresdn** [comment]
@lvm I realized that my config wasn't quite done, so if you'd tried this, please try it again. I cleaned up a bunch to just try and streamline the testing/project for now.

`poetry run tox` now works properly.

As a side note, I ran across errors with the JSONField in `eav/models.py`, so that definitely needs to be tested better, but also probably switched out for [django-jsonfield-backport](https://github.com/laymonage/django-jsonfield-backport). For example, if postgres isn't installed, and you're on Django 3.1, then JSONField defaults to a TextField, which then blows up in `eav.models.Value.value_json` as it has a named parameter `encoder`.

**Comment by Dresdn** [comment]
Lastly, I was going to setup codecov, but realized I don't access to the Settings -> Secrets page to set the CI token. Let me know if you can add that token, or grant me perms.

**Comment by lvm** [comment]
@Dresdn 

Hey!
I have good and bad news:
The Bad: sorry i haven't reviewed this yet. Ideally i planned maintaining this for a project at work but I'm currently working on something else so i didn't had much time to work on this.
The Good: when moved this to jazzband, the goal was to avoid (me as) a bottleneck, so please go ahead and [request to become the lead](https://github.com/jazzband/help/issues/new?assignees=&labels=lead&template=project_lead.md&title=Project+Lead%3A+django-eav2) (this is, of course, if you want to and to continue working on this without me in the way)

**Pull request merged**